src/adapters/RoomsJSONAdapter.js

```diff
@@ -0,0 +1,83 @@
+import {Observable} from 'rxjs';
+
+import RoomsAdapter from './RoomsAdapter';
+
+/**
+ * @typedef RoomJSON
+ * @param {object} datasource An object that contains a set of rooms keyed by ID.
+ * @example
+ * {
+ *  "room-1": {
+ *     "ID": "room-1",
+ *     "title": "title",
+ *     "roomType": "group"
+ *  }
+ * }
+ */
+
+/*
+ * Implements the RoomsAdapter interface with a JSON object as its datasource. See @RoomsJSON
+ */
+export default class RoomsJSONAdapter extends RoomsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.getRoom = this.getRoom.bind(this);
+  }
+
+  /**
+   * Returns an observable that emits room data of the given ID.
+   *
+   * @param {string} ID ID of room to get
+   * @returns {Observable.<Room>}
+   * @memberof RoomsJSONAdapter
+   */
+  getRoom(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find room with ID "${ID}"`));
+      }
+
+      observer.complete();
+    });
+  }
+
+  /**
+   * Returns an observable that emits an array of previous activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<string>>}
+   * @memberof RoomsAdapter
+   */
+  getPreviousRoomActivities(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find previous activities associated to the room with ID "${ID}"`));
```

Comment:
In case of error, we should probably send an empty array. We shouldn't crash the app because there was no data. Instead, the component will render nothing by passing it an empty array..
In addition, we can `console.error` a message for developers if we want.

Reply:
I kinda addressed your comments. take a look